src/flask_ml/flask_ml_server/utils.py

```diff
@@ -0,0 +1,359 @@
+        match parameter_schema.value:
+            case RangedFloatParameterDescriptor():
+                parameters[parameter_schema.key] = parameter_schema.value.range.min
+            case FloatParameterDescriptor():
+                parameters[parameter_schema.key] = parameter_schema.value.default
+            case EnumParameterDescriptor():
+                parameters[parameter_schema.key] = parameter_schema.value.enum_vals[0].key
+            case TextParameterDescriptor():
+                parameters[parameter_schema.key] = parameter_schema.value.default
+            case RangedIntParameterDescriptor():
+                parameters[parameter_schema.key] = parameter_schema.value.range.min
+            case IntParameterDescriptor():
+                parameters[parameter_schema.key] = parameter_schema.value.default
+            case _:
+                assert_never(parameter_schema.value)
+    return RequestBody(inputs=inputs, parameters=parameters)
+
+
+def resolve_input_sample(input_type: Any) -> Input:
+    # TODO: Add a parameterized test for this function
+    match input_type:
+        case models.FileInput:
+            return Input(root=FileInput(path="/Users/path/to/file"))
+        case models.DirectoryInput:
+            return Input(root=DirectoryInput(path="/Users/path/to/folder"))
+        case models.TextInput:
+            return Input(root=TextInput(text="A sample piece of text"))
+        case models.BatchFileInput:
+            return Input(
+                root=BatchFileInput(
+                    files=[
+                        FileInput(path="/Users/path/to/file1"),
+                        FileInput(path="/Users/path/to/file2"),
+                    ]
+                )
+            )
+        case models.BatchTextInput:
+            return Input(
+                root=BatchTextInput(
+                    texts=[
+                        TextInput(text="A sample piece of text 1"),
+                        TextInput(text="A sample piece of text 2"),
+                    ]
+                )
+            )
+        case models.BatchDirectoryInput:
+            return Input(
+                root=BatchDirectoryInput(
+                    directories=[
+                        DirectoryInput(path="/Users/path/to/folder1"),
+                        DirectoryInput(path="/Users/path/to/folder2"),
+                    ]
+                )
+            )
+        case _:
+            assert_never(input_type)
+
+
+def is_typeddict(cls):
+    return isinstance(cls, type) and hasattr(cls, "__annotations__")
+
+
+def ensure_ml_func_parameters_are_typed_dict(ml_function: Callable[[Any, Any], ResponseBody]):
+    hints = get_type_hints(ml_function)
+    if not is_typeddict(hints["inputs"]):
+        raise BadRequestError(f"Inputs must be a TypedDict")
+    if not is_typeddict(hints["parameters"]):
+        raise BadRequestError(f"Parameters must be a TypedDict")
+
+
+def ensure_ml_func_hinting_and_task_schemas_are_valid(
```

Comment:
Function is doing too many things. It's too long. Should be broken up into multiple functions.